### PR TITLE
Parse yaml config and expose settings via getConfiguration() in worker

### PR DESCRIPTION
**Pull request by rtannenbaum**
Make WorkerExtensionContext parse yaml config and expose settings via ExtensionContext.getConfiguration()

ExtensionContext.getConfiguration() is already used in GoogleCloudExtension.

#197

### CODE REVIEW

File: portability-worker/src/main/java/org/dataportabilityproject/worker/ConfigUtils.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+final class ConfigUtils {
+  static Map<String, Object> parse(InputStream in) throws IOException {
+    ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    return mapper.readValue(in, Map.class);
+  }
+}
```
Comment (seehamrun) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/ConfigUtils.java]: nit: new line at EOF

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java [hunk 2]
```diff
@@ -30,16 +33,38 @@
  * {@link ExtensionContext} used by the worker.
  */
 final class WorkerExtensionContext implements ExtensionContext {
+  // TODO: build script, which should take cloud extension name and environment (e.g. local) as
```
Comment (seehamrun) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java]: Does it make sense to move this TODO to the build script?

File: portability-worker/src/test/java/org/dataportabilityproject/worker/ConfigUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    InputStream in = new ByteArrayInputStream(testYaml.getBytes(StandardCharsets.UTF_8));
+    try {
+      Map<String, Object> config = ConfigUtils.parse(in);
+      assertThat(config.get("environment")).isEqualTo("PROD");
+      assertThat(config.get("cloud")).isEqualTo("LOCAL");
+    } catch (IOException e) {
+      fail("Could not parse test yaml. Got exception: " + e);
+    }
+  }
+}
```
Comment (seehamrun) [inline portability-worker/src/test/java/org/dataportabilityproject/worker/ConfigUtilsTest.java]: nit: new line at EOF

File: portability-worker/src/main/java/org/dataportabilityproject/worker/ConfigUtils.java (hunk 1 above)
Comment (rtannenbaum) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/ConfigUtils.java]: this is weird.. I have the setting configured to add a new line in Intellij. But then I have to manually add *another* newline at the bottom for it to add line 31 here. In Intellij, it shows the cursor on line 32.. has that happened to you?

File: portability-worker/src/test/java/org/dataportabilityproject/worker/ConfigUtilsTest.java (hunk 3 above)
Comment (rtannenbaum) [reply portability-worker/src/test/java/org/dataportabilityproject/worker/ConfigUtilsTest.java]: Done

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java (hunk 2 above)
Comment (rtannenbaum) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java]: yes, good point

File: portability-worker/src/main/java/org/dataportabilityproject/worker/ConfigUtils.java (hunk 1 above)
Comment (seehamrun) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/ConfigUtils.java]: Thats odd. I haven't run into this and tried to reproduce with a new class but it looks like its working fine.

Comment (rtannenbaum) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/ConfigUtils.java]: yeah, weird. thanks for trying it out.

**Pull request merged**